ament_cmake_core/cmake/core/all.cmake

```diff
@@ -53,12 +53,23 @@ _set_cmake_policy_to_new_if_available(CMP0020)
 _set_cmake_policy_to_new_if_available(CMP0021)
 _set_cmake_policy_to_new_if_available(CMP0022)
 _set_cmake_policy_to_new_if_available(CMP0023)
+_set_cmake_policy_to_new_if_available(CMP0042)
+
+# set RPATH to ON for OS X
+if(APPLE)
+  set(CMAKE_MACOSX_RPATH ON)
+endif()
 
 # the following operations must be performed inside a project context
 if(NOT PROJECT_NAME)
   project(ament_cmake_internal NONE)
 endif()
 
+# use AMENT_ENABLE_TESTING to avoid warnings about not using it
+if(DEFINED AMENT_ENABLE_TESTING AND ${AMENT_ENABLE_TESTING})
```

Comment:
How is that more future proof? Are you talking about CMP0054? The point is to use, the boolean evaluation doesn't actually matter.